projects/motor_controller_interface/src/motor_controller_fsm.c

```diff
@@ -0,0 +1,321 @@
+#include <string.h>
+
+#include "motor_controller_fsm.h"
+
+#include "can.h"
+#include "can_interval.h"
+#include "can_unpack.h"
+#include "config.h"
+#include "exported_enums.h"
+#include "fifo.h"
+#include "motor_controller.h"
+#include "motor_controller_interface_events.h"
+
+static FSM s_fsm;
+static Fifo *s_fifo;
+static MotorControllerMeasurement *s_mc_measurements;
+static CanInterval *s_left_interval = NULL;
+static CanInterval *s_right_interval = NULL;
+
+// Generic "dumb" handler to copy telemetry values to a buffer
+static void prv_handle_slave_msg(const GenericCanMsg *msg, void *context) {
+  memcpy(context, &msg->data, MIN(msg->dlc, (size_t)8));
+}
+
+static StatusCode prv_push_drive(const CANMessage *msg, void *context, CANAckStatus *ack_reply) {
+  DriverControlsData data = { 0 };
+
+  CAN_UNPACK_MOTOR_CONTROLS(msg, &data.throttle, &data.direction, &data.cruise_control,
```

Comment:
Right now the 4th parameter is the steering angle, but I have a PR open in `codegen-tooling` to change that to the brake state.

Reply:
This change has now been merged in.